### ISSUE TITLE
Better typing for UserClamis

### ISSUE THREAD

**Issue by cafesanu** [okta/okta-auth-js#902]
https://github.com/okta/okta-auth-js/blob/e39e66db2e24df1e1ca31bf082e49095292da043/lib/types/UserClaims.ts#L21

`interface UserClaims` right now has as its last property `[propName: string]: any;` in here https://github.com/okta/okta-auth-js/blob/e39e66db2e24df1e1ca31bf082e49095292da043/lib/types/UserClaims.ts#L40

This allows me to add anything I want without Typescript helping me if there are wrong organization customer claims. Instead I would suggest changing the interface to a type so we can extend a generic.

I would propose the following:

```diff
- export interface UserClaims {
+ export type UserClaims<T = {}> = T & {
    auth_time?: number;
    aud?: string;
    email?: string;
    email_verified?: boolean;
    exp?: number;
    family_name?: string;
    given_name?: string;
    iat?: number;
    iss?: string;
    jti?: string;
    locale?: string;
    name?: string;
    nonce?: string;
    preferred_username?: string;
    sub: string;
    updated_at?: number;
    ver?: number;
    zoneinfo?: string;
-    [propName: string]: any;
}
```

This way if I have custom organizational claims I can do something in my code like:
```ts
export interface MyOrgClaims {
  groups: string[];
}
...
oktaAuth.getUser()
  .then((user: UserClaims<MyOrgClaims>) => {
     const groups = user.groups; // TS won't complain
     const sup = user.sup; // TS will complain!
  });
...

```

Opinions?

**Comment by shuowu** [comment]
@cafesanu Thanks for the suggestion! 

Internal Ref: OKTA-419985

**Comment by cafesanu** [comment]
@shuowu  Happy to open a PR if the Okta team thinks it makes sense

**Comment by shuowu** [comment]
@cafesanu That would be great! Can you also follow the [Contributing guide](https://github.com/okta/okta-auth-js/blob/master/CONTRIBUTING.md) to sign the CLA when the PR is ready? Then our team will review and get it in.

**Comment by shuowu** [comment]
Released in [6.0.0](https://github.com/okta/okta-auth-js/releases/tag/okta-auth-js-6.0.0)
